spec/hardware/main.fmf

```diff
@@ -115,6 +115,27 @@ description: |
 
     __ https://pint.readthedocs.io/
 
+    Names and IDs
+    -------------
+
+    When looking for specific devices, it is often possible to
+    request them either by their name or by some kind of ID. The
+    specification follows a simple namign scheme when a property of
+    a device can be expressed as a name as well as an ID:
+
+    * keys with ``-name`` suffix represent human-comprehensible
+      string names - model name, vendor name, microarchitecture code
+      name, brand names, and so on. For example, ``Intel(R) Xeon(R)
+      CPU E5-2670 v2 @ 2.50GHz`` would be a ``cpu.model-name``
+      requirement.
+    * IDs are left without a suffix. IDs tend to be integers, or
+      groups of integers. For example, ``62`` would be a
+      ``cpu.model`` requirement.
```

Comment:
This suggests that IDs are integers only, yet the schema allows both `integer` or `string`. Why is that?

Reply:
The string alternative is necessary to allow numbers wrapped within quotes, e.g. `model: "12345"` is perfectly acceptable, and it will be converted to an integer by tmt.

IDs *tend* to be integers, but it's not a rule. To allow for weirder IDs, the string is also helpful. Who knows, maybe there are some HW devices with special buses where the device vendor is identified by colon-separated integers, `12:34:43:21`.

Reply:
Yeah, that makes sense.